specification/security/resource-manager/Microsoft.Security/stable/2019-08-01/iotAlertTypes.json

```diff
@@ -0,0 +1,259 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "Security Center",
+    "description": "API spec for Microsoft.Security (Azure Security Center) resource provider",
+    "version": "2019-08-01"
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "security": [
+    {
+      "azure_auth": [
+        "user_impersonation"
+      ]
+    }
+  ],
+  "securityDefinitions": {
+    "azure_auth": {
+      "type": "oauth2",
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Security/iotSecuritySolutions/{solutionName}/analyticsModels/default/iotAlertTypes/{iotAlertTypeName}": {
```

Comment:
You should also have a path to list all alert types.

Reply:
Is this a must-requirement?
If possible, we prefer to not expose a list path, but instead allow the user to query specific ones.

Reply:
Unfortunately, it is. Having a resource comes with certain expectations from the platform even if it's a read only resource. This is documented in the RPC repo at https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/resource-api-reference.md. There are several services in Azure that rely on this to perform various automated scans. Weird deviations like this cause problems for them and also cause customer complaints about API inconsistencies.

Reply:
Added List-AlertTypes path.

Reply:
Thanks!

---
In reply to: [438729472](https://github.com/Azure/azure-rest-api-specs/pull/9653#discussion_r438729472) [](ancestors = 438729472)